src/main/java/seedu/address/model/item/Item.java

```diff
@@ -0,0 +1,98 @@
+package seedu.address.model.item;
+
+import java.util.Objects;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+/**
+ * Represents an Item in FoodREM.
+ * TODO: Implement Item with Tags
+ */
+public class Item {
+    public final ItemName name;
+    public final ItemQuantity quantity;
+    public final ItemDate boughtDate;
+    public final ItemDate expiryDate;
+
+    /**
+     * Constructs an item.
+     * @param name Name of the item.
+     * @param quantity Quantity of the item.
+     * @param boughtDate Date when the item was purchased.
+     * @param expiryDate Date when the item expires.
+     * TODO: Make expiryDate an optional field.
```

Comment:
From my understanding, everything should be optional except for name. Do correct me if I am wrong.